### PR TITLE
fix: correctly pass Action and View options in the Java SDK

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

**Comment by ennru** [comment]
Scala compilation fails https://app.circleci.com/pipelines/github/lightbend/kalix-jvm-sdk/3507/workflows/a70034aa-2ac1-4b13-9179-cdfddb653a94/jobs/43295?invite=true#step-108-1270

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala
Comment (octonato) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala:51]: Maybe we could use `ActionOptions.defaults` instead of wrapping with a `Option`

**Comment by octonato** [comment]
@raboof, I fixed this PR and did a force push on your branch because I rebased it with main.

File: sdk/java-sdk/src/test/scala/kalix/javasdk/impl/action/ActionHandlerSpec.scala
Comment (octonato) [inline sdk/java-sdk/src/test/scala/kalix/javasdk/impl/action/ActionHandlerSpec.scala:74]: For some reason, after adding the `Option[ActionOptions]`, the type inference stopped to work for `ActionFactory`. That's why I'm had to type it explicitly. 🤷

**Review by octonato — APPROVED** [review]
LGTM, but I did one commit in this PR (ea1745634f432d1f9940e223f7bc5e7719268772), so I won't merge it myself.

File: codegen/scala-gen/src/main/scala/kalix/codegen/scalasdk/impl/SourceGenerator.scala
Comment (raboof) [inline codegen/scala-gen/src/main/scala/kalix/codegen/scalasdk/impl/SourceGenerator.scala:142]: I know I did this myself, but looking at it again with fresh eyes it doesn't necessarily make sense to me anymore: if we initialize it with `Some`, does it need to be an `Option` at all?

**Review by raboof — COMMENTED** [review]
Thanks for finishing this up. we should probably verify if the `javaPackageOption` is really needed here, otherwise LGTM.

File: codegen/scala-gen/src/main/scala/kalix/codegen/scalasdk/impl/SourceGenerator.scala
Comment (octonato) [reply codegen/scala-gen/src/main/scala/kalix/codegen/scalasdk/impl/SourceGenerator.scala:142]: I think we should keep it. Or change in another PR in any case. 

As far as I can tell, when we fill it from a descriptor, it can be empty: 
https://github.com/lightbend/kalix-jvm-sdk/blob/c5c537144006d425a3087b28aa9a772dfb0376fe/codegen/core/src/main/scala/kalix/codegen/MessageType.scala#L177-L178

**Pull request merged**
